### PR TITLE
Restore Dashboards: Add e2e tests

### LINKED ISSUE #89526: RestoreDashboards: Add e2e tests for all work flows
**Test the following workflows:**
- in dashboard browse view: dashboard gets selected => click on delete button => check whether the new additional text is available => click delete => appears in Recently Deleted section
- Restoring from Recently Deleted: a dashboard gets selected => click restore button => check whether the number in the modal is right => click restore => toast shows that the dashboard got restored => check in browse dashboard view whether the restored dashboard is back
- Permanently deleting from Recently Deleted: a dashboard gets selected => click permanently delete button => check whether the number in the modal is right => click delete => toast shows that the dashboard got deleted

**Note:**
- Make sure the feat toogle is set to true

### PR DESCRIPTION
**Pull request by JoaoSilvaGrafana**
**What is this feature?**

This PR adds the e2e tests for the Restore Dashboards feature.

**Why do we need this feature?**

To make sure the workflow is working

**Who is this feature for?**

Everyone

**Which issue(s) does this PR fix?**:

Fixes #89526

**Special notes for your reviewer:**

~~Leaving this in draft just to show it works, but afterwards will be removing the feature toggles from the `grafana-server/custom.ini` and skipping the test until the feature is turned on for everyone.~~

Some changes had to be made for other tests to work, making sure dashboards are permanently deleted, not just soft deleted, so if we want to merge these e2e tests at all we either have to merge this PR as is, with the feature flags added to `grafana-server/custom.ini` or we remove the feature flags, skip this test and comment all the other changes from different tests.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: e2e/utils/flows/deleteDashboard.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export const deleteDashboard = ({ quick = false, title, uid }: DeleteDashboardCo
 
 const quickDelete = (uid: string) => {
   cy.request('DELETE', fromBaseUrl(`/api/dashboards/uid/${uid}`));
+  cy.request('DELETE', fromBaseUrl(`/api/dashboards/uid/${uid}/trash`));
```
Comment (L-M-K-B) [inline e2e/utils/flows/deleteDashboard.ts:32]: Why do you need to add this although `deleteDashboard()` is not used in e2e/dashboards/TestRestoreDashboard.json?

File: e2e/dashboards-suite/dashboard-restore.spec.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    // Restore dashboard
+    e2e.pages.Search.table.row('E2E Test - Restore Dashboard').find('[type="checkbox"]').click({ force: true });
+    cy.contains('button', 'Restore').click();
+    cy.contains('p', 'This action will restore 1 dashboard.').should('be.visible');
+    e2e.pages.ConfirmModal.delete().click();
+
+    // Dashboard should appear in Browse
+    e2e.pages.Dashboards.visit();
+    e2e.pages.BrowseDashboards.table.row('E2E Test - Restore Dashboard').should('exist');
+
```
Comment (L-M-K-B) [inline e2e/dashboards-suite/dashboard-restore.spec.ts:30]: It's possible to delete dashboards in a different way. Can you please add a step here where you test the flow when deleting the dashboard from its settings page?

File: e2e/dashboards-suite/dashboard-restore.spec.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    e2e.pages.Dashboards.visit();
+    e2e.pages.BrowseDashboards.table.row('E2E Test - Restore Dashboard').should('not.exist');
+  });
+});
+
+const deleteDashboard = () => {
+  e2e.pages.BrowseDashboards.table.row('E2E Test - Restore Dashboard').find('[type="checkbox"]').click({ force: true });
+
+  cy.contains('button', 'Delete').click();
+  e2e.flows.confirmDelete();
```
Comment (L-M-K-B) [inline e2e/dashboards-suite/dashboard-restore.spec.ts:53]: Is there an opportunity to check whether the Toast is showing the right text?
Screenshot 2024-09-04 at 11 46 27

**Review by L-M-K-B — COMMENTED** [review]
This is awesome.🚀 However I added a couple of comments to make sure the tested workflow is complete.

File: e2e/utils/flows/deleteDashboard.ts (hunk 1 above)
Comment (JoaoSilvaGrafana) [reply e2e/utils/flows/deleteDashboard.ts:32]: There is a panels test that creates a dashboard at the start of each test and deletes it at the end using this method. We need to permanently delete the dashboard so the next test can reimport it

**Review by Clarity-89 — APPROVED** [review]
Looks good!

**Review by L-M-K-B — APPROVED** [review]
🥳

**Pull request merged**